### PR TITLE
Added root cargo workspace

### PR DESCRIPTION
**Pull request by Kixiron**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[workspace]
```
Comment (ryzhyk) [inline Cargo.toml:1]: Would it make more sense to keep this file in `rust/` rather than in the project root?

Comment (Kixiron) [reply Cargo.toml:1]: Either will work, it's up to you

Comment (ryzhyk) [reply Cargo.toml:1]: In that case `rust` is a better location.

Comment (Kixiron) [reply Cargo.toml:1]: Ah, I forgot the entire point of the file, whoops. Being in the root of the project allows language servers to detect it and subsequently check the rust files we want it to, being nested within a folder requires opening that directly with the editor

Comment (ryzhyk) [reply Cargo.toml:1]: Fair enough. Merging as is.  Thanks!

**Pull request merged**